Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -1416,8 +1427,9 @@ void RenderLayerCompositor::traverseUnchangedSubtree(RenderLayer* ancestorLayer,
         computeExtent(overlapMap, layer, layerExtent);
 
     if (layer.paintsIntoProvidedBacking()) {
-        auto* provider = backingSharingState.backingProviderCandidateForLayer(layer, *this, overlapMap, layerExtent);
+        auto* provider = backingSharingState.previousBackingProviderCandidateForLayer(layer);
```

Comment:
Since this could render incorrectly (in situations when we previously crashed), would it be worth considering:
1. Keep backingProviderCandidateForLayer as a first attempt.
2. If nullptr -> debug-only `ASSERT`, then `LOG_ERROR`, and fallback on previousBackingProviderCandidateForLayer.
3. optional: Change to a `RELEASE_ASSERT(provider)` after that, so it will pop out more in crash traces.

Reply:
It's not obvious that doing the two attempts would improve outcomes. If `backingProviderCandidateForLayer` returns a non-null answer, but still different to the previous/existing, then we're going to have incorrect painting.

Logging an error is a good idea, hopefully will be helpful if anyone reports incorrect rendering.